app/src/main/kotlin/com/softteco/template/ui/feature/signUp/SignUpViewModel.kt

```diff
@@ -123,7 +126,7 @@ class SignUpViewModel @Inject constructor(
     @Immutable
     data class State(
         val loading: Boolean = false,
-        val registrationState: Boolean = false,
+        val registrationState: StateHandler = StateHandler.Loading,
```

Comment:
We don't utilize StateHandle as intended because we still have separate load, snackbar, and requestState fields in State